js/blocks/controls/user-input.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * The UserInput component, forked from the URLInput component in Gutenberg.
+ *
+ * It would be ideal to extend that component instead of forking it.
+ * But there are changes throughout this class.
+ * For example, URLInput stores the URL and post, and this only stores the user slug.
+ * This still depends on styling from URLInput, including the class editor-url-input__suggestion.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/0ede174e6ff482085ee51b6a99bea0801c11d609/packages/editor/src/components/url-input/index.js
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import scrollIntoView from 'dom-scroll-into-view';
+
+/**
+ * WordPress dependencies
+ */
+const { __, sprintf, _n } = wp.i18n;
+const { Component, createRef } = wp.element;
+const { decodeEntities } = wp.htmlEntities;
+const { UP, DOWN, ENTER, TAB } = wp.keycodes;
+const { Spinner, withSpokenMessages, Popover } = wp.components;
+const { withInstanceId } = wp.compose;
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+// Since UserInput is rendered in the context of other inputs, but should be
+// considered a separate modal node, prevent keyboard events from propagating
+// as being considered from the input.
+const stopEventPropagation = ( event ) => event.stopPropagation();
+
+class UserInput extends Component {
+	constructor( { autocompleteRef } ) {
+		super( ...arguments );
+
+		this.onChange = this.onChange.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.autocompleteRef = autocompleteRef || createRef();
+		this.inputRef = createRef();
+		this.updateSuggestions = this.updateSuggestions.bind( this );
+
+		this.suggestionNodes = [];
+
+		this.state = {
+			users: [],
+			showSuggestions: false,
+			selectedSuggestion: null,
+		};
+	}
+
+	componentDidUpdate() {
+		const { showSuggestions, selectedSuggestion } = this.state;
+		// only have to worry about scrolling selected suggestion into view
```

Comment:
Hey! This PR is _amazing_. But I challenged myself to find _something_ to review you on. :P

Let's be consistent with our comments – always start with a Capital Letter, and finish with a full stop.

Reply:
Haha, thanks! 

Good point, 2e3f04b9059c1b4096135ff04606bc4bb537645a corrects this.